### ISSUE TITLE
Responsive?

### ISSUE THREAD

**Issue by rohan-deshpande** [reactjs/react-tabs#142]
Great component!

Any thoughts on making these responsive? There's some nice ideas [here](http://stackoverflow.com/questions/25855428/make-bootstrap-3-tabs-responsive) using Bootstrap 3 tabs.

**Comment by dmitry** [comment]
@rohan-deshpande by the time I'm using https://github.com/maslianok/react-responsive-tabs

**Comment by sgilligan** [comment]
my preference is to transform into a drop down rather than accordion (for mobile scenarios) eg https://codepen.io/jpkempf/details/Awxbo

**Comment by dmitry** [comment]
What if dropdown is too long?

**Comment by sgilligan** [comment]
yes, the dropdown would need some smarts to make it scrollable if there were too many elements for the available space.

**Comment by MartinHaeusler** [comment]
I just ran into this issue as well. This library does everything a tabsheet needs to do (and does it nicely) except for proper overflow handling. I guess this has to be some CSS media query magic in the background to make it happen?

I'm fine with either the dropdown or left/right scroll buttons (slight preference for the dropdown). I just would like to avoid:
 - tabs becoming unreachable for the user (except by resizing the window)
 - showing the vanilla browser scrollbar

**Comment by bmsterling** [comment]
For those that are interested, I did fork this repo to make a responsive version similar to what @sgilligan posted: https://github.com/localmotors/react-tabs

You can see in action: https://launchforth.io/styleguide/react/

**Comment by sgilligan** [comment]
nice! exactly what I was looking for. @dmitry if the drop down is long, it wouldn't be any different to the length of an accordion style implementation. I like the implementation from @bmsterling as when selected, the UI is keep minimal which is great on mobile.

**Comment by grawlinson** [comment]
Any plans for forking the changes back into this repo? Would be nice.

**Comment by danez** [comment]
@bmsterling I think this can be done now with react-tabs itself by simply doing.

```
<Tabs>
  <div className="react-tabs__menu is--open">
    <TabList> ... </TabList>
  </div>
  <TabPanel>content</TabPanel>
</Tabs>
```

And it needs the css changes obviously. It would be nice to have an example in the repo that shows how this works, but I currently don't have time to do that.

**Comment by bmsterling** [comment]
@danez thanks, I'll give it a pass this week.

**Comment by Frozzy6** [comment]
is this task still relevant?

**Comment by brunocramos** [comment]
I'm more into horizontal scrolling on mobile, simply adding:
```css
@media (your-own-screen-restrictions) {
  .react-tabs__tab-list {
      display: block;
      overflow-y: auto;
      width: 100%;
      white-space: nowrap;
  }
}
```

**Comment by KoolTheba** [comment]
The last example by @brunocramos i'm afraid is vertical, no horizontal.
I've tried many different ways and there is no way to make `TabList` horizontally-scrolled.
Anything, please, despite almost 3 years have past since the last comment?? 🙏

**Comment by joepvl** [comment]
@KoolTheba you can try the following snippet:
```css
.react-tabs__tab-list {
  display: flex;
  overflow-x: auto;
}
```

**Comment by TechStudent10** [comment]
Are there any plans to build this into the main lib? It's been over a year since the last response.

**Comment by joepvl** [comment]
@TechStudent10 no, it's out of scope. Using some of the code from this thread you should be able to do it yourself without much fuss.

**Comment by jpveilleux** [comment]
I find it hard to believe that in 2023 (even when this thread was created all the way back in 2016) that responsive support is "out of scope". I mean ... you should be mobile first in fact...

**Comment by deflexable** [comment]
@joepvl how do you enable horizontal scroll on ``<TabPanel />``
